core/trino-main/src/main/java/io/trino/sql/planner/PlanOptimizers.java

```diff
@@ -575,6 +578,7 @@ public PlanOptimizers(
                                 .addAll(columnPruningRules)
                                 .add(new InlineProjections(plannerContext, typeAnalyzer))
                                 .addAll(new PushFilterThroughCountAggregation(plannerContext).rules()) // must run after PredicatePushDown and after TransformFilteringSemiJoinToInnerJoin
+                                .addAll(new RemoveFalseFiltersAfterDomainTranslator(plannerContext).rules())
```

Comment:
did you mean to invoke RemoveTrivialFilters as well?

Reply:
As mentioned at https://github.com/trinodb/trino/pull/16206#discussion_r1155313277 I've added this line because I saw `PushPredicateIntoTableScan` at `PlanOptimizers:600` may produce `None` predicates. IMO this realization is insufficient to determine that `RemoveTrivialFilters` invocation is also required. Do you think otherwise?

Reply:
I don't understand. Perhaps you need to explain this to me once more.
I am just curious what's the usefulness of RemoveFalseFiltersAfterDomainTranslator if not coming together with RemoveTrivialFilters. It won't remove the FilterNode.

Reply:
You are right, now that `RemoveFalseFiltersAfterDomainTranslator` is an `ExpressionRewriteRuleSet` and not a standalone rule, `RemoveTrivialFilters` is also required. Added.